static/app/components/feedback/feedbackItem/useResolveFeedback.tsx

```diff
@@ -0,0 +1,85 @@
+import {Fragment, useCallback, useMemo} from 'react';
+
+import {
+  addErrorMessage,
+  addLoadingMessage,
+  addSuccessMessage,
+} from 'sentry/actionCreators/indicator';
+import {ModalRenderProps, openModal} from 'sentry/actionCreators/modal';
+import {Button} from 'sentry/components/button';
+import {useInfiniteFeedbackListData} from 'sentry/components/feedback/feedbackDataContext';
+import {Flex} from 'sentry/components/profiling/flex';
+import {t, tct} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {HydratedFeedbackItem} from 'sentry/utils/feedback/item/types';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+import useUrlParams from 'sentry/utils/useUrlParams';
+
+interface Props {
+  feedbackItem: HydratedFeedbackItem;
+}
+
+function openResolveModal({
+  onResolve,
+  label,
+}: {
+  label: string;
+  onResolve: () => void | Promise<void>;
+}) {
+  openModal(({Body, Footer, closeModal}: ModalRenderProps) => (
+    <Fragment>
+      <Body>{tct('[label] this feedback?', {label})}</Body>
+      <Footer>
+        <Flex gap={space(1)}>
+          <Button onClick={closeModal}>{t('Cancel')}</Button>
+          <Button
+            priority="primary"
+            onClick={() => {
+              closeModal();
+              onResolve();
+            }}
+          >
+            {label}
+          </Button>
+        </Flex>
+      </Footer>
+    </Fragment>
+  ));
+}
+
+export default function useResolveFeedback({feedbackItem}: Props) {
+  const feedbackId = feedbackItem.feedback_id;
+
+  const api = useApi();
+  const organization = useOrganization();
+  const {setParamValue: setFeedbackSlug} = useUrlParams('feedbackSlug');
+  const {setFeedback} = useInfiniteFeedbackListData();
+
+  const url = useMemo(() => {
+    return `/organizations/${organization.slug}/issues/${feedbackId}/`;
+  }, [feedbackId, organization]);
+
+  const handleResolve = useCallback(async () => {
+    addLoadingMessage(t('Updating feedback...'));
+    try {
+      await api.requestPromise(url, {
+        method: 'PUT',
+        data: {status: feedbackItem.status === 'unresolved' ? 'resolved' : 'unresolved'},
+      });
+      addSuccessMessage(t('Updated feedback'));
+      setFeedbackSlug('');
```

Comment:
We dont' need this one, that's what makes the UI go back to "nothing is selected" but for archive/unarchive we can keep looking at the same item, as long as the values are up to date.